### DISCUSSION TITLE
About Incremental Strategy

### DISCUSSION THREAD

**Discussion by MichalisDBA** [osalvador/ReplicaDB#81] [Ideas]
Hello.

I read about [Incremental](https://osalvador.github.io/ReplicaDB/docs/docs.html#incremental) mode and i want to ask.

Wouldn't be better have an option, that as long as you can connect to both source and sink databases and there is ability to Free-form queries, for incremental sync to do something like this:

select * from source.db.table where id (column) not in (select id (column) from sink.db.table)

So that you do a query to the sink table, get the data and then do a query to the source table excluding the data from the sink table.

The final result set can just be inserted to the sink table of your choice.

**Comment by osalvador** [comment]
Hi @MichalisDBA,

In this type of query you need a sql query processor, i.e. a database. So ReplicaDB should replicate all the data from the sink to the source database to compare both... This is a huge effort, and maybe in a TB data table, impossible. 

Data virtualization tools, like Denodo, can process a query from different databases but i don't aware any one open source.
